### PR TITLE
 PP 10210 yourpsp stripe display tasklist

### PR DESCRIPTION
**Pull request by AbdirizakIdris**
- Updated yourPSP get controller
- Added unit test for get controller including tasklist and KYC conditions

### CODE REVIEW

File: app/controllers/your-psp/get.controller.test.js [hunk 1]
```diff
@@ -64,4 +67,78 @@ describe('Your PSP GET controller', () => {
     expect(pageData.enableStripeOnboardingTaskList).to.equal(false)
     sinon.assert.calledWith(res.render, 'your-psp/index')
   })
+
+  it('should get stripe taskList and for your PSP', async () => {
```
Comment (kbottla) [inline app/controllers/your-psp/get.controller.test.js:71]:
```suggestion
  it('should get stripe taskList when requiresAdditionalKycData is false', async () => {
```

File: app/controllers/your-psp/get.controller.test.js [hunk 2]
```diff
@@ -64,4 +67,78 @@ describe('Your PSP GET controller', () => {
...
+      ENTER_COMPANY_NUMBER: { enabled: true, completed: false },
+      CONFIRM_ORGANISATION_DETAILS: { enabled: true, completed: false },
+      UPLOAD_GOVERNMENT_ENTITY_DOCUMENT: { enabled: false, completed: false }
+    })
+    expect(pageData.taskListIsComplete).to.equal(false)
+    expect(Object.keys(pageData.taskList).length).to.equal(7)
+    sinon.assert.calledWith(res.render, 'your-psp/index')
+  })
+
+  it('should get KYC tasks when requiresAdditionalKycData is set to true', async () => {
```
Comment (kbottla) [inline app/controllers/your-psp/get.controller.test.js:91]:
```suggestion
  it('should get KYC tasks when requiresAdditionalKycData is true', async () => {
```

File: app/controllers/your-psp/get.controller.test.js (hunk 1 above)
Comment (iqbalgds) [inline app/controllers/your-psp/get.controller.test.js:71]: I would recommend changing to `should get stripe taskList when using Stripe`

File: app/controllers/your-psp/get.controller.test.js [hunk 3]
```diff
@@ -64,4 +67,78 @@ describe('Your PSP GET controller', () => {
...
+
+    req.account.requires_additional_kyc_data = true
+
+    const getTaskListMock = sinon.stub().resolves({
```
Comment (iqbalgds) [inline app/controllers/your-psp/get.controller.test.js:95]: Can you please format so it is more condensed?

File: app/controllers/your-psp/get.controller.test.js [hunk 4]
```diff
@@ -64,4 +67,78 @@ describe('Your PSP GET controller', () => {
...
+      }
+    })
+
+    const controller = getControllerWithMocks(getTaskListMock)
+
+    await controller(req, res, next)
+
+    const pageData = res.render.args[0][1]
+
+    expect(pageData.kycTaskList).to.deep.equal({
```
Comment (iqbalgds) [inline app/controllers/your-psp/get.controller.test.js:116]: Can you please format so it is more condensed?

**Review by iqbalgds — CHANGES_REQUESTED** [review]
Looks great - just some minor suggestions.

File: app/controllers/your-psp/get.controller.test.js (hunk 1 above)
Comment (AbdirizakIdris) [reply app/controllers/your-psp/get.controller.test.js:71]: sounds good

File: app/controllers/your-psp/get.controller.test.js [hunk 5]
```diff
@@ -64,4 +67,78 @@ describe('Your PSP GET controller', () => {
...
+    expect(Object.keys(pageData.taskList).length).to.equal(7)
+    sinon.assert.calledWith(res.render, 'your-psp/index')
+  })
+
+  it('should get KYC tasks when requiresAdditionalKycData is set to true', async () => {
+
+    req.account.requires_additional_kyc_data = true
+
+    const getTaskListMock = sinon.stub().resolves({
+      'ENTER_ORGANISATION_URL': {
```
[the remark this answers is not in the document]
Comment (AbdirizakIdris) [reply app/controllers/your-psp/get.controller.test.js:96]: Good idea

**Review by iqbalgds — DISMISSED** [review]

**Review by iqbalgds — APPROVED** [review]
I'm confident that this change does not require further review and meets the secure coding requirements of GOV.UK Pay

**Pull request merged**